src/hotspot/cpu/aarch64/aarch64_neon.ad

```diff
@@ -181,6 +181,73 @@ instruct reinterpretX2D(vecD dst, vecX src)
   ins_pipe(vlogical64);
 %}
 
+
+instruct reinterpretStoX(vecX dst, vecD src)
+%{
+  predicate(n->bottom_type()->is_vect()->length_in_bytes() == 16 &&
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes() == 4);
+  match(Set dst (VectorReinterpret src));
+  ins_cost(INSN_COST);
+  format %{ " # reinterpret $dst,$src\t# S to X" %}
+  ins_encode %{
+    // If registers are the same, no register move is required - the
+    // upper bits of "src" are expected to have been initialized
+    // to zero.
```

Comment:
I have a little concern about this assumption. How to ensure the upper bits are zero? Since the `ReinterpretNode` could be used separately and not always with `CastNode` together. E.g

https://github.com/openjdk/jdk/blob/jdk-18%2B9/src/hotspot/share/opto/vectorIntrinsics.cpp#L844

Reply:
There are several places in the AArch64 back end where we expect the upper bits of a register to be zero, but we've never depended on it. This is not a good time to start, so let's clear the bits in order to be certain.

Reply:
This is the same as reinterpretD2X. The upper 96 bits of "src" are zero,  I guess it is because when the 128-bit register is initialized, the upper 96 bits will be zeroed, and only its lower 32-bit data will be manipulated later.

Reply:
OK, I'll clear the higher bits in the next commit.